src/CSET/operators/constraints.py

```diff
@@ -84,29 +84,33 @@ def generate_level_constraint(
     ---------
     coordinate: str
         Level coordinate name about which to constraint.
-    levels: int | list[int]
-        CF compliant levels.
+    levels: int | list[int] | str
+        CF compliant levels, or ``"*"`` for retrieving all levels available.
 
     Returns
     -------
     constraint: iris.Constraint
     """
-    # Ensure is iterable.
-    if not isinstance(levels, Iterable):
-        levels = [levels]
+    # If astericks, then return all levels for given coordinate.
+    if levels == "*":
+        return iris.Constraint(**{coordinate: lambda cell: True})
+    else:
+        # Ensure is iterable.
+        if not isinstance(levels, Iterable):
+            levels = [levels]
 
-    # When no levels specified reject cube with level coordinate.
-    if len(levels) == 0:
+        # When no levels specified reject cube with level coordinate.
+        if len(levels) == 0:
 
-        def no_levels(cube):
-            # Reject cubes for which coordinate exists.
-            return not bool(cube.coords(coordinate))
+            def no_levels(cube):
+                # Reject cubes for which coordinate exists.
+                return not bool(cube.coords(coordinate))
```

Comment:
```suggestion
                return not cube.coords(coordinate)
```
Not automatically casts to a boolean.